packages/core/src/render3/instructions/change_detection.ts

```diff
@@ -279,34 +289,43 @@ function detectChangesInComponent(
     hostLView: LView, componentHostIdx: number, mode: ChangeDetectionMode): void {
   ngDevMode && assertEqual(isCreationMode(hostLView), false, 'Should be run in update mode');
   const componentView = getComponentLViewByIndex(componentHostIdx, hostLView);
-  detectChangesInView(componentView, mode);
+  detectChangesInViewIfAttached(componentView, mode);
 }
 
 /**
  * Visits a view as part of change detection traversal.
  *
- * - If the view is detached, no additional traversal happens.
+ * If the view is detached, no additional traversal happens.
+ */
+function detectChangesInViewIfAttached(lView: LView, mode: ChangeDetectionMode) {
+  if (viewAttachedToChangeDetector(lView)) {
```

Comment:
Stylistic suggestion:
```suggestion
  if (!viewAttachedToChangeDetector(lView)) {
    return;
  }
  // ...
```

Early-exit rather than nesting here would make it more clear what the "main flow" is.